### ISSUE TITLE
[BUG] `timeout-minutes` should allow expressions (but it demands `floats`)

### ISSUE THREAD

**Issue by webknjaz** [rhysd/actionlint#384]
The same problem has just been fixed in the jsonschema based linter: https://github.com/python-jsonschema/check-jsonschema/issues/354.

TL;DR the field should be either an expression or an integer. I don't think GHA actually allows floats in there like the error implies:
```console
$ pre-commit run actionlint-docker --all-files -v 
Lint GitHub Actions workflow files.......................................Failed
- hook id: actionlint-docker
- duration: 1.03s
- exit code: 1

.github/workflows/reusable-build-wheel.yml:42:22: type of expression at "float number value" must be number but found type string [expression]
   |
42 |     timeout-minutes: ${{ inputs.qemu && 60 || 20 }}
   |                      ^~~
```

**Comment by rhysd** [comment]
```yaml
on: push
jobs:
  test:
    runs-on: ubuntu-latest
    timeout-minutes: ${{ true && 60 || 20 }}
    steps:
      - run: echo hello
```

I tried this workflow and actionlint reported no error. What is the type of `inputs.qemu`?

**Comment by webknjaz** [comment]
It's a string, so it's falsy when empty:
https://github.com/aio-libs/yarl/blob/bd5ff24/.github/workflows/reusable-build-wheel.yml#L26

**Comment by rhysd** [comment]
Thanks. I could reproduce it.

```yaml
on: push
jobs:
  test:
    runs-on: ubuntu-latest
    timeout-minutes: ${{ '...' && 60 || 20 }}
    steps:
      - run: echo hello
```
